### PR TITLE
Chris/restful api

### PR DESCRIPTION
**Pull request by hehachris**
- [x] Configure app entrypoint to load routes from config
- [x] GET `/pets/{id}`
- [x] POST `/pets`
- [x] GET `/pets/{id}/matches`
- [x] POST `/customers`
- [x] GET `/customers/{id}`
- [x] GET `/customers/{id}/matches`
- [x] POST `/customers/{id}/adopt?pet_id={pet_id}`

### CODE REVIEW

File: app/repos/sequelize/pet.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+        if (customerPreference.age !== null) {
+            criteria.age = {
+                $lte: customerPreference.age
+            };
+        }
+
+        if (customerPreference.species !== null) {
+            criteria.species = {
+                $in: customerPreference.species.split(',')
```
Comment (anthonylau) [inline app/repos/sequelize/pet.js:17]: is not a enum?

File: app/repos/sequelize/pet.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            };
+        }
+
+        if (customerPreference.breed !== null) {
+            criteria.breed = {
+                $in: customerPreference.breed.split(',')
```
Comment (anthonylau) [inline app/repos/sequelize/pet.js:23]: is not a enum?

File: app/repos/sequelize/pet.js (hunk 1 above)
Comment (hehachris) [reply app/repos/sequelize/pet.js:17]: Thanks. Fixed in #13

File: app/repos/sequelize/pet.js (hunk 2 above)
Comment (hehachris) [reply app/repos/sequelize/pet.js:23]: Thanks. Fixed in #13

**Pull request merged**